### PR TITLE
source: handle case when `gitRecorder` isn't configured on an instance

### PR DESCRIPTION
**Pull request by BolajiOlajide**
This PR fixes [a situation](https://github.com/sourcegraph/sourcegraph/pull/55210#issuecomment-1647804475) where the `gitRecorder` configuration isn't set on a Sourcegraph instance.

## Test plan

* Ensure `gitRecorder` isn't set in your site configuration
* Access any repository settings page and click on the `Logs` link
* You should be able to view the `Logs` page and a message saying there are no recorded commands for this repository.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/recorded_commands_test.go [hunk 1]
```diff
@@ -24,6 +25,45 @@ func TestRecordedCommandsResolver(t *testing.T) {
...
+	backend.Mocks.Repos.GetByName = func(context.Context, api.RepoName) (*types.Repo, error) {
+		return &types.Repo{Name: api.RepoName(repoName)}, nil
+	}
+	t.Cleanup(func() {
+		backend.Mocks = backend.MockServices{}
+	})
+
+	repos := database.NewMockRepoStore()
+	repos.GetFunc.SetDefaultReturn(&types.Repo{Name: api.RepoName(repoName)}, nil)
+	db.ReposFunc.SetDefaultReturn(repos)
```
Comment (mrnugget) [inline cmd/frontend/graphqlbackend/recorded_commands_test.go:40]: You need both? `backend.Mocks.Repos.GetByName` and `db.Repos().Get()`? For this test?

Comment (sashaostrikov) [reply cmd/frontend/graphqlbackend/recorded_commands_test.go:40]: Agree, only the site config mock will do

**Review by sashaostrikov — APPROVED** [review]
LGTM! Agree with Thorsten about mocks in tests

File: cmd/frontend/graphqlbackend/recorded_commands_test.go (hunk 1 above)
Comment (BolajiOlajide) [reply cmd/frontend/graphqlbackend/recorded_commands_test.go]: Nope, just mumbled everything together. I just pushed a commit to separate it.

I only need `backend.Mocks.Repos.GetByName` for this test and not `db.Repos().Get()`

**Pull request merged**